test/specs/scripts/extract.spec.js

```diff
@@ -36,7 +42,30 @@ describe('extract', () => {
       expect(messages).not.toContain('msgid "Not in the result"')
     })
   })
+  
+  describe('replacements', () => {
+    const replacementsDir = `${fixtureDir}/replacements`
+
+    afterEach(() => {
+      removeIfExists(`${replacementsDir}/messages.pot`)
+    })
 
+    it('should be possible to add replacement translations based on the replacements json', () => {
+      expect(existsSync(`${replacementsDir}/messages.pot`)).toBeFalsy()
+
+      callForDirReplacements(replacementsDir)
+
+      expect(existsSync(`${replacementsDir}/messages.pot`)).toBeDefined()
+      
+      const messages = readFileSync(`${replacementsDir}/messages.pot`).toString("utf-8")
+
+      expect(messages).toContain('msgid "Needs replacement"')
+      expect(messages).toContain(`msgid "A replacement for 'Needs replacement'`)
+      expect(messages).toContain('#. REPLACEMENT')
+      expect(messages).toContain('msgid "No replacement is needed"')
+    })
```

Comment:
**Praise:** Great test 🔆 👍🏻 

**Suggestion:** Would you agree it would also be good to test whether the `i18n` also picks up the replacement translations?